docs/content/troubleshooting/log_status_collect.md

```diff
@@ -1,41 +1,108 @@
-# Log and status collection
+# Detecting errors and log collection
 
 Use the CSI (Container Storage Interface) driver debug information for problem identification.
 
 **Note:** These procedures are applicable for both Kubernetes and Red Hat® OpenShift®. For Red Hat OpenShift, replace `kubectl` with `oc` in all relevant commands.
 
-To collect and display status and logs related to the different components of IBM® block storage CSI driver, use the following Kubernetes commands:
+## Detecting errors
+
+To help pinpoint potential causes for stateful pod failure:
+
+1.  Verify that all CSI pods are running.
+    ```
+    kubectl get pods -n <namespace> -l csi
+    ```
+
+2.  If a pod is not in a _Running_ state, run the following command:
+    ```
+    kubectl describe -n <namespace> pod/<pod-name>
+    ```
+    View the logs.
+
+## Status and log collection
+To collect and display status and logs related to the different components of IBM® block storage CSI driver, use the Kubernetes found in this section.
+
+Before you begin collecting logs, at the master host level, create a directory for the logs.
+
+For example:
+
+```
+mkdir logs
+```
+Save logs and status reports directly to the created directory by adding in the `> logs/<log_filename>` syntax to the end of collection command.
+
+**Important:** When gathering logs from the storage system, be sure that the logs cover any relevant time frames for the specific issues that you are trying to debug.
+
+### General status and log collection for IBM Support
+When engaging with IBM Support, be sure to run the following steps and copy the output to an external file, when sending log collections:
+
+1. Check the node status.
+    
+    `kubectl get nodes`
+2. Check the CSI driver component status.
+
+    `kubectl get -n <namespace> pod -o wide | grep ibm-block-csi`
+3. Check if the PersistentVolumeClaims (PVCs) are _Bound_.
+
+    `kubectl get pvc`
+
+    - If the PVCs are not bound, (in the _XXX_ state) collect the events of all unbound PVCs. (See [Log collection for unbound PVCs](#log-collection-for-unbound-pvcs).)
 
-## Status collection for CSI pods, daemonset, and statefulset
+### Log collection for all pods and containers
+To collect logs for all pods and containers, use the following commands:
 
+    nodepods=`kubectl get -n <namespace> pod -l app.kubernetes.io/component=csi-node --output=jsonpath={.items..metadata.name}`
+    
+    for pod in $nodepods; do  kubectl logs -n <namespace> --all-containers $pod > logs/$pod; done
```

Comment:
I don't think we should encourage users to put the logs from all the pod's containers into a single file

Reply:
so you want us to just not put this one in at all?

Reply:
@oriyarde  @roysahar  please confirm

Reply:
@rivkap I would write:
for pod in $nodepods;do for container in `kubectl get -n default pod $pod -o jsonpath='{.spec.containers[*].name}'`;do kubectl logs -n default $pod -c $container > /tmp/logs_collection/${pod}_${container}.log;done;done

Reply:
@roysahar  updated according to our meeting and your input
DONE - please approve

Reply:
I think we need [test confirmation](https://github.com/IBM/ibm-block-csi-driver/pull/407#discussion_r766512218) (specifically here, and in general), before approving

Reply:
@oriyarde 
I updated this file together with @roysahar . who else needs to test this section? In any case, testing can be done from the finalized docs_release-1.8.0 branch

Reply:
I don't think @roysahar had a "test" ticket. I meant that he should have one in addition to his review.
I agree testing can be done from the finalized branch, but IMO it would be better to split the effort into separate PRs, separate test tickets, etc...

Reply:
FYI opened [PR 418](https://github.com/IBM/ibm-block-csi-driver/pull/418) for updates

Reply:
thanks!